### PR TITLE
[Form] Support Translatable Enum

### PR DESCRIPTION
**Pull request by Seb33300**
Documentation for https://github.com/symfony/symfony/pull/50931

### CODE REVIEW

File: reference/forms/types/enum.rst [hunk 1]
```diff
@@ -51,17 +51,34 @@ these values as ``<input type="checkbox">`` or ``<input type="radio">``.
 
 The label displayed in the ``<option>`` elements of the ``<select>`` is the enum
 name. PHP defines some strict rules for these names (e.g. they can't contain
-dots or spaces). If you need more flexibility for these labels, use the
-``choice_label`` option and define a function that returns the custom label::
+dots or spaces). If you need more flexibility for these labels, your enum can
+implements ``TranslatableInterface`` to translate or display custom labels::
```
Comment (derrabus) [inline reference/forms/types/enum.rst]:
```suggestion
implement ``TranslatableInterface`` to translate or display custom labels::
```

**Review by derrabus — CHANGES_REQUESTED** [review]

**Comment by derrabus** [comment]
Code PR has been merged.

**Comment by xabbuh** [comment]
We also need a `versionadded` directive which explains that you can use this feature since Symfony 6.4.

**Comment by Seb33300** [comment]
Added

File: reference/forms/types/enum.rst [hunk 2]
```diff
@@ -80,6 +80,10 @@ implement ``TranslatableInterface`` to translate or display custom labels::
         }
     }
 
+.. versionadded:: 6.4
+
+    Support for ``TranslatableInterface`` was introduced in Symfony 6.4.
```
Comment (derrabus) [inline reference/forms/types/enum.rst:85]: The interface itself has existed before 6.4. Maybe we should be a bit more precise here.

```suggestion
    Support for enums that implement ``TranslatableInterface`` was introduced in Symfony 6.4.
```

Comment (Seb33300) [reply reference/forms/types/enum.rst:85]: Symfony (twig) already supports enums that implement `TranslatableInterface`.
But EnumType did not support them yet.

This is the EnumType documentation page so I think it's obvious that we are talking about the EnumType here.

> Support for ``TranslatableInterface`` was introduced in Symfony 6.4.

means

> Support for ``TranslatableInterface`` (in ``EnumType``) was introduced in Symfony 6.4.

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
Thanks for your work on this new feature!

**Pull request merged**